protocols/horizon/operations/main.go

```diff
@@ -401,15 +401,14 @@ type RestoreFootprint struct {
 //
 // Amount - expressed as a signed decimal to 7 digits precision.
 // Asset  - the classic asset expressed as issuer and code.
-// DestinationMuxedId - If the event's destination was a muxed address, then include the that in the response as a string
+// DestinationMuxedId - If the event's destination was a muxed address, then include the uint64 memoId in the response as a string
 type AssetContractBalanceChange struct {
 	base.Asset
-	Type                   string `json:"type"`
-	From                   string `json:"from,omitempty"`
-	To                     string `json:"to,omitempty"`
-	Amount                 string `json:"amount"`
-	DestinationMuxedIdType string `json:"destination_muxed_id_type,omitempty"`
-	DestinationMuxedId     string `json:"destination_muxed_id,omitempty"`
+	Type               string `json:"type"`
+	From               string `json:"from,omitempty"`
+	To                 string `json:"to,omitempty"`
+	Amount             string `json:"amount"`
+	DestinationMuxedId string `json:"destination_muxed_id,omitempty"`
```

Comment:
Should this be
```suggestion
	DestinationMuxedId uint64 `json:"destination_muxed_id,string,omitempty"`
```
or does that cause problems in the "memo set to zero" case due to the `omitempty`?

Reply:
I actually did know that `DestinationMuxedId uint64` `json:.....string` was something you could do

Reply:
Yeah it's a clever trick since it lets you work with the native value on the Go side, but in this case I think if you were to receive a memo with a value of zero, you'd end up omitting it. So you'd actually have to make it a pointer (`*uint64`) which makes the code way more unnecessarily complicated, so I think keeping it as a string internally (since there's no semantic value Go-side in having it be a uint64 [no math or comparisons being done, etc.]) is better here.

Reply:
That's an interesting observation re: existing code, though... I wonder if it's different in the case of `MuxedAccount` vs. `Memo` usage. Thoughts @tamirms?

Reply:
the only recourse then is to make it a *string then

Reply:
If it's a `string` on the Go side, it'll be `""` in the absence of a memo, and be set to a `"%d"` in the ID case, right? So you don't need a pointer. Basically I'm saying I think it's good as-is but wanted to bring up the discussion point for visibility & understanding.

Reply:
Actually, it still makes sense to have it just as "DestinationMuxedId string `json:"destination_muxed_id,omitempty"`"

When saving it as a jsonblob in DB, I save it as a string.
https://github.com/stellar/go/pull/5739/files/08098be92cd0a361ea9399db00eb4eb0e68415b1#diff-5f37fbbec67583bbaa94cbe645d8a4aa046505cee2c37be660818ef340645ab1R820

So, even if the actualMemo set in the event is uint64(0), it will still be captured correctly/stored in DB as a string,  and show up in the restapi response.

Tl;dr - IMO, it makes sense to retain the existing logic and not change it

Reply:
Even better - yep, agreed :+1: